backend/controllers/Event.js

```diff
@@ -65,7 +85,96 @@ const createEvent = async (req, res) => {
  * req: request object containing user id and event id.
  * res: response object to return status 200.
  */
+/**
+ * NOTES:
+ * - might implement time caped. Rigth now, expected that start and end time is in the same day
+ * - anyone can click finish for now
+ */
+
+const finishEvent = async (req, res) => {
+  try {
+    const eventId = req.params.id;
+
+    // the point will only be distributed if user finish the event before the end time
+
+    const event = await Event.findById(eventId);
+
+    // if event is completed, wont do anything
+    if (event.status === 'completed') {
+      res.status(400).json({ message: 'Event already completed' });
+    }
+
+    // get the point based on the time difference between start and end time
+    let period = event.end.getTime() - event.start.getTime();
+    period = period / 1000 / 60 / 60; // convert from milisecond to hours
+    let point = period * 10;
+
+    // check if the event is already overdue and deduct point accordingly
+    const currentTime = new Date();
+    if (event.end.getTime() < currentTime.getTime()) {
+      point = Math.floor(point / 2);
+    }
+
+    // Find EventParticipation of the users with that event ID and status == accepted,
+    // then for each Event Participation, update the point of that user
+
+    const participation = await EventParticipation.find({
+      eventId: eventId,
+      status: 'accepted',
```

Comment:
enum would be good here. So you can't misspell on your bad day even if you want to.